### PR TITLE
Highlight Mint code in Markdown content.

### PR DESCRIPTION
**Pull request by gdotdesign**
This PR implements highlighting for Mint language blocks in Markdown content in here documents.

### CODE REVIEW

**Comment by farism** [comment]
This will be useful for static docs generator too.

File: src/utils/markd_vdom_renderer.cr [hunk 1]
```diff
@@ -69,7 +69,30 @@ module Mint
...
-      @io << '`' << node.text.gsub('`', "\\`").strip << '`'
+
+      if language == "mint"
+        parser = Parser.new(node.text, "source.mint")
+        parser.parse_any
+
+        parts =
+          SemanticTokenizer.tokenize(parser.ast)
+
+        parts.map_with_index do |item, index|
```
Comment (Sija) [inline src/utils/markd_vdom_renderer.cr:80]:
```suggestion
        parts.each_with_index do |item, index|
```

File: src/utils/markd_vdom_renderer.cr [hunk 2]
```diff
@@ -69,7 +69,30 @@ module Mint
...
+            tag("span", {"className" => item[0].to_s.underscore})
+            @io << '`' << item[1].gsub('`', "\\`") << '`'
+            tag_end
+          end
+
+          @io << ','
+        end
+      else
+        @io << '`' << node.text.gsub('`', "\\`").strip << '`'
+      end
```
Comment (Sija) [inline src/utils/markd_vdom_renderer.cr:94]: I'd suggest extracting this into a dedicated method:

```cr
def inline_code(io, code)
  io << '`' << code.gsub('`', "\\`") << '`'
end
```

```suggestion
      if language == "mint"
        parser = Parser.new(node.text, "source.mint")
        parser.parse_any

        parts =
          SemanticTokenizer.tokenize(parser.ast)

        parts.each do |item|
          case item
          in String
            inline_code(@io, item)
          in Tuple(SemanticTokenizer::TokenType, String)
            tag("span", {"className" => item[0].to_s.underscore})
            inline_code(@io, item[1])
            tag_end
          end

          @io << ','
        end
      else
        inline_code(@io, node.text.strip)
      end
```

**Comment by gdotdesign** [comment]
This PR is now outdated and will be superseded by the compiler refactor in #680 and will close when that's merged.

**Comment by gdotdesign** [comment]
Closing, since this is included in the `0.20.0` release.

**Pull request closed without merging**